src/components/trends/smbg/FocusedSMBGRangeLabels.js

```diff
@@ -0,0 +1,101 @@
+ * == BSD2 LICENSE ==
+ */
+
+import React, { PropTypes } from 'react';
+import Tooltip from '../common/Tooltip';
+
+import { MGDL_UNITS, MMOLL_UNITS } from '../../../utils/constants';
+import { displayBgValue } from '../../../utils/format';
+import { THREE_HRS, millisecondsAsTimeOfDay } from '../../../utils/datetime';
+
+import styles from './FocusedSMBGRangeLabels.css';
+
+const ONE_AND_A_HALF_HOURS = THREE_HRS / 2;
+
+const FocusedSMBGRangeLabels = (props) => {
+  const { focusedRange } = props;
+  if (!focusedRange) {
+    return null;
+  }
+
+  const { bgUnits, focusedRange: { data, position } } = props;
+  const timeFrom = millisecondsAsTimeOfDay(data.msX - ONE_AND_A_HALF_HOURS);
+  const timeTo = millisecondsAsTimeOfDay(data.msX + ONE_AND_A_HALF_HOURS);
+  const maxPosition = { top: position.yPositions.max, left: position.left };
+  const meanPosition = { top: position.yPositions.mean, left: position.left };
+  const meanSide = position.tooltipLeft ? 'left' : 'right';
+  const minPosition = { top: position.yPositions.min, left: position.left };
+  return (
+    <div className={styles.container}>
+      <Tooltip
+        content={<span className={styles.number}>{displayBgValue(data.max, bgUnits)}</span>}
+        position={maxPosition}
+        side={'top'}
+        tail={false}
+        offset={{ top: -5, left: 0 }}
+      />
+      <Tooltip
+        title={<span className={styles.explainerText}>{timeFrom} - {timeTo}</span>}
+        content={
+          <span className={styles.number}>
+            Average {displayBgValue(data.mean, bgUnits)}
+          </span>
+        }
+        side={meanSide}
+        position={meanPosition}
+        offset={{ top: 0, left: position.tooltipLeft ? -10 : 10 }}
+      />
+      <Tooltip
+        content={<span className={styles.number}>{displayBgValue(data.min, bgUnits)}</span>}
+        position={minPosition}
+        side={'bottom'}
+        tail={false}
+        offset={{ top: 5, left: 0 }}
+      />
+    </div>
+  );
+};
+
+FocusedSMBGRangeLabels.defaultProps = {};
+
+FocusedSMBGRangeLabels.propTypes = {
+  bgUnits: PropTypes.oneOf([MGDL_UNITS, MMOLL_UNITS]).isRequired,
+  focusedRange: PropTypes.shape({
+    data: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      max: PropTypes.number.isRequired,
+      mean: PropTypes.number.isRequired,
+      min: PropTypes.number.isRequired,
+      msX: PropTypes.number.isRequired,
+    }).isRequired,
+    position: PropTypes.shape({
+      left: PropTypes.number.isRequired,
+      tooltipLeft: PropTypes.bool.isRequired,
+      yPositions: PropTypes.shape({
+        max: PropTypes.number.isRequired,
+        mean: PropTypes.number.isRequired,
+        min: PropTypes.number.isRequired,
+      }).isRequired,
+    }).isRequired,
+  }),
+  timePrefs: PropTypes.shape({
+    timezoneAware: React.PropTypes.bool.isRequired,
+    timezoneName: React.PropTypes.oneOfType([React.PropTypes.string, null]),
```

Comment:
Have you done a pass through on all the code (outside of just smbg trends) as requested above? Would love it if we're confident we're improving things as we go.